alpha/lib/model/asset.php

```diff
@@ -547,6 +547,22 @@ public function getDownloadUrl($useCdn = false, $forceProxy = false, $preview =
 
 		return $this->getDownloadUrlWithExpiry(86400, $useCdn, $forceProxy, $preview);
 	}
+
+	public function getServeFlavorUrl($entry)
+	{
+		if ($entry->getType() != entryType::MEDIA_CLIP)

```

Comment:
Document entry may also have assets, and media entry assets might be thumbnails, captions and related files that shouldn't use serve-flavor.
Maybe this method should be placed in flavorAsset and not in asset.